### PR TITLE
network-rpc: Implement a new network rpc layout with proc macro

### PR DESCRIPTION
**Pull request by fikgol**
1.  Add a new create for all the network rpcs.
2. Generate rpc client && server by the rpc specification trait.
3. ReImplement the network in sync helpers.

### CODE REVIEW

File: network-rpc/core/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "network-rpc-core"
+version = "0.1.0"
+authors = ["fikgol <hit.cs.lijun@gmail.com>"]
```
Comment (jolestar) [inline network-rpc/core/Cargo.toml:4]: 这个统一成  ["Starcoin Core Dev <dev@starcoin.org>"] 吧

File: network/src/network.rs [hunk 2]
```diff
@@ -505,12 +505,12 @@ impl Inner {
                     .await?;
             }
 
-            PeerMessage::RawRPCRequest(id, _rpc_path, request) => {
+            PeerMessage::RawRPCRequest(id, rpc_path, request) => {
```
Comment (jolestar) [inline network/src/network.rs:508]: PeerMessage::RawRPCRequest 是不是可以考虑删除然后直接用  RpcClient 调用了？

**Comment by jolestar** [comment]
Add Copywrite and license to the source files header.

File: network/src/network.rs (hunk 2 above)
Comment (fikgol) [reply network/src/network.rs:508]: 能把现在 sync/process.rs里的 Requests 删掉。RawRpcRequest 是更底层的逻辑。

**Pull request merged**